### PR TITLE
Create a clean compiling version of rtl_433 for Windows using the Vis…

### PR DESCRIPTION
**Pull request by ms-jdow**
…ual Studio 17 Community Edition (free) compiler.

Signed-off-by: msjdow <msjdow@users.noreply.github.com>

### CODE REVIEW

**Comment by zuckschwerdt** [comment]
I'm not fond of using PCH. Perhaps split that off as optional second patch set. The VS17 looks good.
Good hint on the intmax_t stuff. I need to rethink that and perhaps get rid of it.

I guess removing the latest changes in flex got in by accident, add "upstream", pull, rebase and force push, please:

- git remote add upstream https://github.com/merbanan/rtl_433.git
- git checkout master
- git pull upstream master
- git checkout VS17
- git rebase master
- git push -f

**Comment by zuckschwerdt** [comment]
Looks like you forgot to include `fineoffset_wh31.c` on your last commit.

**Comment by merbanan** [comment]
fineoffset_WH25 seems to use the same protocol as the WH31. Can you confirm that or supply some sample signals?

File: src/devices/flex.c [hunk 1]
```diff
@@ -387,7 +263,11 @@ r_device *flex_create_device(char *spec)
         fprintf(stderr, "Bad flex spec, missing name!\n");
         usage();
     }
+#ifdef WIN32
+    params->name = _strdup(c);
+#else
     params->name = strdup(c);
+#endif
     snprintf(dev->name, sizeof(dev->name), "General purpose decoder '%s'", c);
```
Comment (merbanan) [inline src/devices/flex.c]: Riddling the code with #ifdefs does not help maintenance. I'd rather have one file that contains all the platform defines.

File: src/devices/hondaremote.c [hunk 2]
```diff
@@ -48,7 +47,7 @@ static int hondaremote_callback(bitbuffer_t *bitbuffer) {
 
 //call function to lookup what button was pressed
 			const char* code = get_command_codes(bytes);
-			device_id = bytes[44]<<8 | bytes[45];
+			device_id = (bytes[44]<<8|bytes[45]);
```
Comment (merbanan) [inline src/devices/hondaremote.c]: Does that cause a warning in VS17?

**Comment by ms-jdow** [comment]
On 2018-12-07 11:42, Benjamin Larsson wrote:
> *@merbanan* commented on this pull request.
> 
> --------------------------------------------------------------------------------
> 
> In src/devices/flex.c 
> <https://github.com/merbanan/rtl_433/pull/818#discussion_r239924939>:
> 
>>      params->name = strdup(c);
> +#endif
>       snprintf(dev->name, sizeof(dev->name), "General purpose decoder '%s'", c);
> 
> Riddling the code with #ifdefs does not help maintenance. I'd rather have one 
> file that contains all the platform defines.

That seems like a good idea to me. I was interested in making it compile in a hurry.

{^_-}   Joanne

**Comment by zuckschwerdt** [comment]
Good discussions and some interesting ideas, a few got picked up already some will take more time/deliberation. Closing now to tidy PRs up.

**Pull request closed without merging**